bluemira/base/reactor.py

```diff
@@ -421,16 +498,44 @@ def _construct_xyz_cad(
                 degree=(360 / self.n_sectors) * n_sectors,
             )
 
-    def show_cad(
+    def _filter_and_reconstruct(
+        self,
+        dims_to_show: Tuple[str, ...],
+        with_components: Optional[List[ComponentManager]],
+        n_sectors: Optional[int],
+        filter_: Union[Callable[[Component], bool], None],
+        **kwargs,
+    ):
+        # We filter because self.component (above) only creates
+        # a new root node for this reactor, not a new component tree.
+        comp_copy = self._filter_tree(
+            self.component(with_components), dims_to_show, filter_
+        )
+        # if "xyz" is requested, construct the 3d cad
+        # from each xyz component in the tree,
+        # as it's assumed that the cad is only built for 1 sector
+        # and is sector symmetric, therefore can be patterned
+        if "xyz" in dims_to_show:
+            self._construct_xyz_cad(
+                comp_copy,
+                with_components,
+                self.n_sectors if n_sectors is None else n_sectors,
+            )
+        return comp_copy
+
+    def save_cad(
         self,
         *dims: str,
         with_components: Optional[List[ComponentManager]] = None,
         n_sectors: Optional[int] = None,
         filter_: Union[Callable[[Component], bool], None] = FilterMaterial(),
+        filename: Optional[str] = None,
+        formatt: Union[str, cadapi.CADFileType] = "stp",
+        directory: str = "",
```

Comment:
`formatt` is used quite widely in the code and if we change we should do it everywhere (also would lead to api breakage elsewhere).
`filter_` is less widely used (just here and `components.py`), it again should be done everywhere. The only reason is because thats the name `anytree` uses for filtering.

I'm open to both changes but they should be done everywhere. I feel that is outside the scope of this PR. Would you be satified if I open an issue for 2.0?

Reply:
I agree, it should be done everywhere. An issue is good 👍🏼